File: src/main/java/com/dafttech/terra/engine/gui/anchors/GUIAnchor.scala
package com.dafttech.terra.engine.gui.anchors

import com.dafttech.terra.engine.gui.GUIObject
import com.dafttech.terra.engine.gui.containers.GUIContainer

abstract class GUIAnchor {
  def applyAnchor(gObj: GUIObject, container: GUIContainer)
  def needsApplyOnFrame: Boolean = false
  def isContainerDependent: Boolean = true
}

File: src/main/java/com/dafttech/terra/engine/gui/anchors/GUIAnchor.java
package com.dafttech.terra.engine.gui.anchors;

import com.dafttech.terra.engine.gui.GUIObject;
import com.dafttech.terra.engine.gui.containers.GUIContainer;

public abstract class GUIAnchor {
    public abstract void applyAnchor(GUIObject object, GUIContainer container);

    public boolean needsApplyOnFrame() {
        return false;
    }

    public boolean isContainerDependent() {
        return true;
    }
}
